elastic/kibana
# [Synthetics] add monitor id migration

## Summary

Relates to https://github.com/elastic/kibana/issues/143309

Adds a migration to add two fields to each monitor saved object

1. `[ConfigKey.HEARTBEAT_ID] ('id')` - The id to use for ES queries to heartbeat. Either the saved object id for UI monitors, or the custom heartbeat id for project monitors. This new key can be used regardless of monitor type as the single source of truth as the id to use for ES queries
2. `[ConfigKey.CONFIG_ID] ('config_id')` - This id is always the saved object id. It is an exact match to `savedObject.id`. I've added this key as a way to perform aggregations against the monitor saved object id in the future, as it's currently impossible to do terms aggregations against `savedObject.id` but possible to do aggregations os any field within the savedObject attributes. By moving a copy of the id into the saved object attributes, we can perform aggregations if needed in the future.  

This PR is a prerequisite for standardizing the use of `ConfigKey.HEARTBEAT_ID` across the app. Making use of these new keys in a predictable, standardized way will take place in a subsequent PR.

### Testing

1. Checkout the 8.5 branch before pulling down this PR. (Note, you'll want to do this while running ES locally, instead of using a edge oblt ES. If your edge oblt ES is on an older ES version than 8.6, the migration won't go through)
2. Create 1 of each monitor for both UI and project monitors
4. Checkout this PR
5. Navigate to the Monitor Management page and find the get request for monitors
6. View each monitor's `id` and `config_id` value. For project monitors, the `id` value should match `custom_heartbeat_id`. For ui monitors, the `id` value should match the saved object id. For both monitor types, the `config_id` value should match the saved object id. 
<img width="855" alt="Screen Shot 2022-10-27 at 1 20 25 PM" src="https://user-images.githubusercontent.com/11356435/199029263-c6c7b356-8826-4820-b5c8-c0189dcb077f.png">
